astroid/inference.py

```diff
@@ -57,13 +63,11 @@ def infer_end(self, context=None):
     yield self
 
 
-# We add ignores to all these assignments in this file
-# See https://github.com/python/mypy/issues/2427
-nodes.Module._infer = infer_end  # type: ignore[assignment]
-nodes.ClassDef._infer = infer_end  # type: ignore[assignment]
-nodes.Lambda._infer = infer_end  # type: ignore[assignment]
-nodes.Const._infer = infer_end  # type: ignore[assignment]
-nodes.Slice._infer = infer_end  # type: ignore[assignment]
```

Comment:
The `type: ignore[assignment]` comments should still be necessary due to the mypy bug.

Reply:
Yeah, sadly that ignore is also catching all incorrect assignments (difference in types).

Just wanted to show that this doesn't create any actual typing issues.